ibm/resource_ibm_cos_bucket.go

```diff
@@ -434,6 +472,99 @@ func resourceIBMCOSUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	//// Update  the Retention policy
```

Comment:
can we move the logic of s3 client intialization outside if block because it some common in many if blocks